.github/workflows/ci-schedule-compatibility.yaml

```diff
@@ -4,6 +4,12 @@ on:
     # Run this workflow "At 20:00 UTC on Sunday and Saturday"
     - cron: '0 20 * * 0,6'
 
+permissions:
+  actions: write # Used by nick-fields/retry@v3.0.0 to retry job steps if needed.
+  contents: read # Required by actions/checkout@v4 to fetch the repository contents.
+  deployments: write # Needed to manage deployments for setting up the test environment.
+  statuses: read # Necessary to check the status of deployments before running tests.
```

Comment:
@aditya7302 Can you explain the reason for adding these two permissions?

Reply:
@zhzhuang-zju 
deployments: write ->This permission is necessary to manage deployments for setting up the test environment. We need to update the images which involves setting up new images. This requires write access to deployments.
```
- name: change kube-apiserver and kube-controller-manager version
  run: |
    kubectl --kubeconfig=${HOME}/.kube/karmada.config --context=karmada-host \
      set image deployment/karmada-apiserver -nkarmada-system \
      karmada-apiserver=registry.k8s.io/kube-apiserver:${{ matrix.kubeapiserver-version }}
    kubectl --kubeconfig=${HOME}/.kube/karmada.config --context=karmada-host \
      set image deployment/karmada-kube-controller-manager -nkarmada-system \
      kube-controller-manager=registry.k8s.io/kube-controller-manager:${{ matrix.kubeapiserver-version }}
```

statuses: read ->This permission is essential to check the status of deployments before running tests.
```
kubectl --kubeconfig=${HOME}/.kube/karmada.config --context=karmada-host \
  rollout status deployment/karmada-kube-controller-manager -nkarmada-system --timeout=5m
kubectl --kubeconfig=${HOME}/.kube/karmada.config --context=karmada-host \
  rollout status deployment/karmada-apiserver -nkarmada-system --timeout=5m
```

Reply:
Refer to https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs, `permission` `deployment` and `statused` are not actually a k8s concept. We might don't need those two permissions. Can you remove those two permissions and try it out？

Reply:
@zhzhuang-zju here is the APIServer compatibility workflow run without ```deployments: write``` and ```statuses: read``` permissions : ```https://github.com/aditya7302/karmada/actions/runs/9694680710```.

Reply:
I've tried this locally and had similar results to yours, a couple of jobs would fail. Might have to configure out the reason for that

Reply:
@aditya7302 It looks like it's all failing at step `change kube-apiserver and kube-controller-manager version`. @liangyuanpeng  is trying to remove that step, so we can wait for his part of the work to be done before defining top level permission for `ci -schedule-compatibility workflow`, what do you think?

Reply:
I also think that some jobs are failing at step ```change kube-apiserver and kube-controller-manager version```. I will wait for @liangyuanpeng to remove this step and until then I will look for other good-first and help-wanted issues to work on.

Reply:
Thhis is pending with https://github.com/karmada-io/karmada/issues/4929 before open a PR to working for update remove the step `change kube-apiserver and kube-controller-manager version`

Reply:
kindly ping @aditya7302 
It seems @zhzhuang-zju is suggesting removing some permissions as per https://github.com/karmada-io/karmada/pull/5068#discussion_r1655700915.